tests/test_resource_warning.py

```diff
@@ -0,0 +1,47 @@
+import asyncio
+import time
+from datetime import datetime
+
+import aiohttp
+
+
+async def _resource_warning(recwarn):
+    timeout = 3 * 60
+
+    request_interval = 60
+    request_increase = 30
+    next_request = 0
+
+    async with aiohttp.ClientSession() as client:
```

Comment:
something like this should work
```suggestion

    async def handler_coro(_request):
        return aiohttp.web.Response(text='Test TLS response')
    http_server = aiohttp.test_utils.RawTestServer(handler_coro, ssl= ssl_ctx)
    await http_server.start_server()

    url = http_server.make_url('/')

    async with aiohttp.ClientSession(
        connector=aiohttp.TCPConnector(ssl=client_ssl_ctx),
    ) as client:
```